packages/trader/src/App/Components/Elements/VideoStream/volume-control.tsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import { CSSTransition } from 'react-transition-group';
+import { Icon } from '@deriv/components';
+
+type TVolumeControl = {
+    onVolumeChange: (new_value: number) => void;
+    volume?: number;
+    is_mobile?: boolean;
+    toggleMute: (new_value: boolean) => void;
+};
+
+const VolumeControl = ({ onVolumeChange, volume, is_mobile, toggleMute }: TVolumeControl) => {
+    const [show_volume, setShowVolume] = React.useState(false);
+    const [is_muted, setIsMuted] = React.useState(false);
+    const [shift_Y, setShiftY] = React.useState<number>(0);
+
+    const volume_bar_ref = React.useRef<HTMLDivElement>(null);
+    const is_dragging = React.useRef<boolean>(false);
+
+    const calculateNewHight = (e: React.MouseEvent<HTMLElement, MouseEvent>) => {
+        const volume_bar = document.querySelector('.player__volume-bar');
+
+        let new_height =
+            (((volume_bar?.getBoundingClientRect().bottom ?? 0) - e.clientY - shift_Y) /
+                (volume_bar?.getBoundingClientRect().height ?? 0)) *
+            100;
+
+        if (new_height >= 100) {
+            new_height = 100;
+        }
+        if (new_height <= 0) {
+            new_height = 0;
+        }
+        return new_height;
+    };
+
+    const checkVolumeControl = (volume: number) => {
+        if (!volume) {
+            setIsMuted(true);
+            toggleMute(true);
+
+            return;
+        }
+        setIsMuted(false);
+        toggleMute(false);
+    };
+
+    const buttonClickHandler = () => {
+        if (is_muted) {
+            volume_bar_ref?.current && volume_bar_ref.current.style.setProperty('height', `${(volume ?? 0.5) * 100}%`);
+            toggleMute(false);
+        } else {
+            volume_bar_ref?.current && volume_bar_ref.current.style.setProperty('height', '0%');
+            toggleMute(true);
+        }
+        setIsMuted(prev => !prev);
+    };
+
+    const mouseMoveHandler = (e: React.MouseEvent<HTMLElement, MouseEvent>) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        if (!is_dragging.current) return;
+        if (!volume_bar_ref?.current) return;
+
+        const new_height = calculateNewHight(e);
+
+        volume_bar_ref.current.style.setProperty('transition', 'none');
+        volume_bar_ref.current.style.setProperty('height', `${new_height}%`);
+
+        onVolumeChange(new_height / 100);
+        checkVolumeControl(new_height);
+    };
+
+    const mouseUpHandler = (e: React.MouseEvent<HTMLElement, MouseEvent>) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        if (!is_dragging.current) return;
+
+        if (volume_bar_ref?.current) {
```

Comment:
```suggestion
        if (volume_bar_ref?.current) {
```
volume_bar_ref won't be undefined